### PR TITLE
B 20631 upgrade go 1.23.1 nodejs 18.20.4 int

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary

- Bring main to int
- Bump go to `1.23.1`
- Bump golangci-lint to [new version](https://github.com/golangci/golangci-lint/releases/tag/v1.61.0) compatible with the latest go
  - Resolve new linter rules
- Bump mockery to [latest](https://github.com/vektra/mockery/releases) to support new go version
  - Adjust pathlib dep to be compatible
  - Add `godebug` to `go.mod` as a temporary workaround to Mockery generation issue https://github.com/vektra/mockery/issues/803
- Bump nodejs to `18.20.4` (LTS)
- Bump CI/CD container to https://github.com/transcom/circleci-docker/pull/471

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

## Related article
The godebug placed in `go.mod` is brand new functionality that came with `1.23`
https://go.dev/doc/godebug

Upgrade guide
https://transcom.github.io/mymove-docs/docs/backend/guides/how-to/upgrade-go-version

### CODE REVIEW

File: .circleci/config.yml
Comment (traskowskycaci) [inline .circleci/config.yml:1]: Wondering if the v7-server-tests-coverage on line 1556 should be v8? cc @deandreJones

Comment (deandreJones) [reply .circleci/config.yml:1]: what cam has is fine

**Review by traskowskycaci — CHANGES_REQUESTED** [review]
Still seeing go-version: '1.22.2' in `.github/workflows/go-auto-approve.yml`

Had a question about v7-server-tests-coverage in .circleci/config.yml - resolved

Only requesting changes for visibility, feel free to dismiss review if no changes needed

**Comment by cameroncaci** [comment]
> Still seeing go-version: '1.22.2' in `.github/workflows/go-auto-approve.yml`
> 
> Had a question about v7-server-tests-coverage in .circleci/config.yml - resolved
> 
> Only requesting changes for visibility, feel free to dismiss review if no changes needed

Great catch, thank you. Resolved now

**Review by traskowskycaci — APPROVED** [review]
Looks good!

**Review by TevinAdams — APPROVED** [review]

**Pull request merged**